### ISSUE TITLE
Future Request: Search and filter data with Amplify.API without custom GraphQL Document

### ISSUE THREAD

**Issue by MarlonJD** [aws-amplify/amplify-flutter#4189]
### Description

Hello, on amplify-flutter there is no search option like ModelQueries.list, do we planning to add this?
I would like to try to add this if it's possible. 

Query predicate operations for search will be one of these:
```
matchPhrasePrefix, matchPhrase, match, multiMatch, ne, range, regexp, eq, exists, gt, gte, lt, lte, wildcard
```

for example search operation can be like this example,
it can be build on the ModelQueries.list,

example ModelQueries.list was like this:
```
final request = ModelQueries.list(
  Student.classType,
  where: Student.NAME.eq("John"),
);
```

to 

```
// eq example
final request = ModelQueries.search(
  Student.classType,
  where: Student.NAME.eq("John"),
  limit: 10,
);

// on search matchPhrasePrefix operation will be important
final request = ModelQueries.search(
  Student.classType,
  where: Student.NAME.matchPhrasePrefix("Jo"),
  limit: 10,
);
```

### Categories

- [ ] Analytics
- [ ] API (REST)
- [X] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [ ] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Platforms

- [X] iOS
- [X] Android
- [X] Web
- [X] macOS
- [X] Windows
- [X] Linux

### Flutter Version

3.3.10

### Amplify Flutter Version

0.6.10

### Deployment Method

Amplify CLI

### Schema

```GraphQL
type Student @model @searchable {
  name: String
  dateOfBirth: AWSDate
  email: AWSEmail
  examsCompleted: Int
}
```

**Comment by Equartey** [comment]
Hi @MarlonJD, thanks for sharing the proactive solution. 

It's unclear to me why you would need a separate `ModelQueries.serach()` method. Could you please explain why your use case could not be solved with a new query predicate passed to the existing `ModelQueries.list()` method?

**Comment by MarlonJD** [comment]
Hi @Equartey, thanks. It's for @searchable directive, see https://docs.amplify.aws/flutter/build-a-backend/graphqlapi/search-and-result-aggregations/ here, it's creating new query in graphql, for example `listTodos`is for normal query, but if you use OpenSearch and get search results for this with `searchTodos`. If I use listTodos, 

1. I have to create customDocument and replace listTodos with searchTodos in created document in `ModelQueries.list()`
2. and there is no `matchPhrasePrefix` in Query predicate operations

So `ModelQueries.search()` creating required variables and correct document for this. So it's solving this issue

**Comment by MarlonJD** [comment]
@Equartey yes I'm doing just like that. So this feature request is for avoid these long and custom codes, that's why I'm proposing this. BTW if you check the PR codes, please let me know what's your ideas about these, and I can change commit message and merge into 1 commit if it's necessary

**Comment by Equartey** [comment]
@MarlonJD Great, yeah I can see the value add for sure. I'll let you know if we want to combine them, thanks.

**Comment by MasterNeuromancer** [comment]
I would like to add that the amplify flutter docs have a dedicated section for how to add search using the @searchable directive, but there is no easy way to access this on the flutter side. It seems like elasticsearch resources are being deployed in that case for no reason. Also that whole section has javascript examples instead of flutter/dart code which is just kinda sad.
